src/components/NcUserStatusIcon/NcUserStatusIcon.vue

```diff
@@ -0,0 +1,173 @@
+  - License, or (at your option) any later version.
+  -
+  - This program is distributed in the hope that it will be useful,
+  - but WITHOUT ANY WARRANTY; without even the implied warranty of
+  - MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+  - GNU Affero General Public License for more details.
+  -
+  - You should have received a copy of the GNU Affero General Public License
+  - along with this program. If not, see <http://www.gnu.org/licenses/>.
+  -
+-->
+
+<docs>
+### Description
+
+This component displays a user status icon.
+
+### Example
+
+```vue
+<template>
+	<div class="row">
+		<NcUserStatusIcon status="online" />
+		<NcUserStatusIcon status="away" />
+		<NcUserStatusIcon status="dnd" />
+		<NcUserStatusIcon status="invisible" />
+	</div>
+</template>
+
+<style>
+.row {
+	display: flex;
+	gap: 10px;
+}
+</style>
+```
+</docs>
+
+<template>
+	<span v-if="activeStatus" v-bind="attributes">
+		<!-- This icon is part of Material UI Icons. Copyright 2020 Google Inc., Apache-2.0 License -->
+		<svg viewBox="0 0 16 16" xmlns="http://www.w3.org/2000/svg">
+			<path v-if="activeStatus === 'online'" fill="var(--color-success)" d="M4.8 11.2h6.4V4.8H4.8v6.4zM8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z" />
+			<template v-else-if="['away', 'busy'].includes(activeStatus)">
+				<path fill="none" d="M-4-4h24v24H-4z" />
+				<path fill="var(--color-warning)" d="M6.9.1C3 .6-.1 4-.1 8c0 4.4 3.6 8 8 8 4 0 7.4-3 8-6.9-1.2 1.3-2.9 2.1-4.7 2.1-3.5 0-6.4-2.9-6.4-6.4 0-1.9.8-3.6 2.1-4.7z" />
+			</template>
+			<template v-else-if="activeStatus === 'dnd'">
+				<path fill="none" d="M-4-4h24v24H-4V-4z" />
+				<path fill="var(--color-error)" d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8z" />
+				<path fill="#fdffff" d="M5 6.5h6c.8 0 1.5.7 1.5 1.5s-.7 1.5-1.5 1.5H5c-.8 0-1.5-.7-1.5-1.5S4.2 6.5 5 6.5z" />
+			</template>
+			<template v-else-if="['invisible', 'offline'].includes(activeStatus)">
+				<path fill="none" d="M-4-4h24v24H-4V-4z" />
+				<path d="M8 0C3.6 0 0 3.6 0 8s3.6 8 8 8 8-3.6 8-8-3.6-8-8-8zm0 3.2c2.7 0 4.8 2.1 4.8 4.8s-2.1 4.8-4.8 4.8S3.2 10.7 3.2 8 5.3 3.2 8 3.2z" />
+			</template>
+		</svg>
+	</span>
+</template>
+
+<script>
+import Vue from 'vue'
+import axios from '@nextcloud/axios'
+import { generateOcsUrl } from '@nextcloud/router'
+
+import { getUserStatusText } from '../../utils/UserStatus.ts'
+import { t } from '../../l10n.js'
+
+export default {
+	name: 'NcUserStatusIcon',
+
+	props: {
+		/**
+		 * Set the user id to fetch the status
+		 */
+		user: {
+			type: String,
+			default: null,
+		},
+
+		/**
+		 * Set the status
+		 */
+		status: {
+			type: String,
+			default: null,
+			validator: (status) => [
+				'online',
+				'away',
+				'busy',
+				'dnd',
+				'invisible',
+				'offline',
+			].includes(status),
+		},
+
+		/**
+		 * Set the `aria-hidden` attribute
+		 */
+		ariaHidden: {
+			type: String,
+			default: null,
+		},
```

Comment:
I don't understand in which case this icon should and should not be `aria-hidden`...
Could you clarify it?

Reply:
See this commit 1ccc246c which migrates NcIconSvgWrapper to NcUserStatusIcon. Since this new component does not need an explicit `name`/aria-label, as it derives it from the `activeStatus` automatically, we need to allow aria-hiding this explicitly which is needed for https://github.com/nextcloud-libraries/nextcloud-vue/blob/91c44a3b8fba7d18dcf0b6ba48862bac0fe55328/src/components/NcAvatar/NcAvatar.vue#L397-L398